client/verta/verta/registry/_check_model_dependencies.py

```diff
@@ -60,14 +61,26 @@ def _check_model_dependencies(
         check_model_dependencies(model_cls=MyModelClass, environment=env)
 
     """
+    env_packages: Set[str] = {parse_req_spec(e)[0] for e in environment.requirements}
     detected_modules: Set[str] = md.class_module_names(model_cls)
-    detected_packages: Set[str] = md.package_names(detected_modules)
-    env_packages: Set[str] = { parse_req_spec(e)[0] for e in environment.requirements }
+    detected_packages: Dict[str, List[str]] = md.package_names(detected_modules)
+
+    missing_packages: Dict[str, List[str]] = {
+        mod: pkgs
+        for mod, pkgs in detected_packages.items()
+        if not env_packages.intersection(set(pkgs))
```

Comment:
Trying to remove potentially unnecessary nesting because this comprehension is already quite complex.

```suggestion
        if not env_packages.intersection(pkgs)
```

Reply:
Ah, I did not realize you could just use `intersection` with a list.  Thanks